backend/app/controllers/products_controller.rb

```diff
@@ -0,0 +1,48 @@
+class ProductsController < ApplicationController
+
+    #GET /products
+    def index
+        @products = Product.all
+        
+        render json: @products
+    end
+
+    #GET /products/1
+    def show 
+        render json: @product
+    end
+
+    #POST /products
+    def create 
+        @product = Product.new(product_params)
+        if @product.save
+            render json: @gproduct, status: :created, location: @product
+        else 
+            render json: @product.errors, status: :unprocessable_entity
+        end
+    end
+
+    #PUT/PATCH /products/1
+    def update
+        if @product.update(product_params)
+            render json: @product
+        else 
+            render json: @product.errors, status: :unprocessable_entity
+        end
+    end
+
+    #DELETE /products/1
+    def destroy
+        @product.destroy
+    end
+
+    private
+    #Use callbacks to share common setup or constrains between actions.
+    def set_product
+        @product = product.find(params[:id])
+    end
+
+    #only allow a list of trusted parameters though.
+    def group_params
+        params.require(:product).permit(:price, :portion_weight_grams, :ingredients, :desription, :image, :name)
```

Comment:
Are you sure it is require a product instance to create or update a product instance?